seleniumbase/core/browser_launcher.py

```diff
@@ -649,12 +649,144 @@ def uc_gui_write(driver, text):
+    return (viewport_x, viewport_y)
+
+
+def uc_gui_click_x_y(driver, x, y, timeframe=0.25, uc_lock=True):
+    install_pyautogui_if_missing(driver)
+    import pyautogui
+    pyautogui = get_configured_pyautogui(pyautogui)
+    screen_width, screen_height = pyautogui.size()
+    if x > screen_width or y > screen_height:
+        raise Exception(
+            "PyAutoGUI cannot click on point (%s, %s)"
+            " outside screen. (Width: %s, Height: %s)"
+            % (x, y, screen_width, screen_height)
+        )
+    if uc_lock:
+        gui_lock = fasteners.InterProcessLock(
+            constants.MultiBrowser.PYAUTOGUILOCK
+        )
+        with gui_lock:  # Prevent issues with multiple processes
+            pyautogui.moveTo(x, y, timeframe, pyautogui.easeOutQuad)
+            if timeframe >= 0.25:
+                time.sleep(0.0555)  # Wait if moving at human-speed
+            if "--debug" in sys.argv:
+                print(" <DEBUG> pyautogui.click(%s, %s)" % (x, y))
+            pyautogui.click(x=x, y=y)
+    else:
+        # Called from a method where the gui_lock is already active
+        pyautogui.moveTo(x, y, timeframe, pyautogui.easeOutQuad)
+        if timeframe >= 0.25:
+            time.sleep(0.0555)  # Wait if moving at human-speed
+        if "--debug" in sys.argv:
+            print(" <DEBUG> pyautogui.click(%s, %s)" % (x, y))
+        pyautogui.click(x=x, y=y)
+
+
+def on_a_cf_turnstile_page(driver):
     source = driver.get_page_source()
     if (
-        "//challenges.cloudflare.com" not in source
-        and 'aria-label="Cloudflare"' not in source
+        "//challenges.cloudflare.com" in source
+        or 'aria-label="Cloudflare"' in source
     ):
+        return True
+    return False
+
+
+def uc_gui_click_cf(driver, frame="iframe", retry=False, blind=False):
+    if not on_a_cf_turnstile_page(driver):
+        return
+    install_pyautogui_if_missing(driver)
+    import pyautogui
+    pyautogui = get_configured_pyautogui(pyautogui)
+    x = None
+    y = None
+    gui_lock = fasteners.InterProcessLock(
+        constants.MultiBrowser.PYAUTOGUILOCK
+    )
+    with gui_lock:  # Prevent issues with multiple processes
+        needs_switch = False
+        is_in_frame = js_utils.is_in_frame(driver)
+        if is_in_frame and driver.is_element_present("#challenge-stage"):
+            driver.switch_to.parent_frame()
+            needs_switch = True
+            is_in_frame = js_utils.is_in_frame(driver)
+        if not is_in_frame:
+            # Make sure the window is on top
+            page_actions.switch_to_window(
+                driver, driver.current_window_handle, 2, uc_lock=False
+            )
+        if not is_in_frame or needs_switch:
+            # Currently not in frame (or nested frame outside CF one)
+            try:
+                i_x, i_y = get_gui_element_position(driver, "iframe")
+                driver.switch_to_frame(frame)
```

Comment:
```suggestion
                i_x, i_y = get_gui_element_position(driver, frame)
                driver.switch_to_frame(frame)
```

to make it work with the specified frame